exp/capacity-planning.md

```diff
@@ -0,0 +1,96 @@
+
+The main purpose of GPU slots is to control the number of containers that are launched on a node, which reduces contention for GPU resources.
+[/note]
+
+<a name="overcommitting"></a>
+## Over-committing resources
+
+If the unused resources on a cluster node don't suffice to launch a container for an application with its defined resource requirements, the container cannot be launched. This behaviour is very restrictive, and in many cases unnecessary.
+
+Usually, a container doesn't use its dedicated CPU cores and memory at 100% all the time. Therefore, AMS allows over-committing available resources. By default, AMS uses a CPU allocation rate of `4` and a memory allocation rate of `2`, which means that it allows four times the number of vCPU cores and twice the amount of RAM per node. See [Configure allocation rates](tbd#configure-allocation-rates) for instructions on how to define the allocation rates for a node.
+
+For example, consider an application that uses the `a2.3` instance type, which requires 2 CPU cores and 3 GB of memory, and you have a node with 8 CPU cores and 16 GB of memory. Without over-commitment, you could only launch four containers before you run out of resources on the node. However, with a CPU allocation rate of `4` and a memory allocation rate of `2` (the default), the available resources on the node change to `4 * 8 CPU cores = 32 CPU cores` and `2 * 16 GB memory = 32 GB memory`, which will allow up to ten containers on the node.
+
+Which CPU allocation rate makes sense depends on the type of application that you are running and what amount of CPU it needs. For low CPU-intensive applications a higher and for high CPU-intensive applications a lower allocation rate makes sense.
+
+## Application requirements
+
+To realistically estimate the required capacity for your deployment, you must consider the type of application that you're running and the expected usage behaviour.
+
+You should [run benchmarks](https://discourse.ubuntu.com/t/how-to-run-benchmarks/17770) to test your application performance and fine-tune the best node and application configuration. Also consider whether your containers use a hardware or software [video encoder](https://discourse.ubuntu.com/t/application-manifest/24197#video-encoder) for video encoding, and what frame rate and resolution they require.
+
+Another aspect is, of course, how many users you expect and how many containers will be running simultaneously. If you expect the usage to be rather consistent, you might not need to plan for huge peeks in load. On the other hand, you must also consider the impact if your cluster runs out of resources and it is not possible anymore to start more containers.
+
+## An example calculation
```

Comment:
What hasn't been clear to me when first reading the example is that we look at the capacity needed not from a per-node perspective but much rather for an overall cluster which has multiple nodes. Can we make that clearer?

Reply:
Added a paragraph to (hopefully :wink: ) make that clearer.